mobile/native/src/storage.rs

```diff
@@ -0,0 +1,215 @@
+use crate::backup::RemoteBackupClient;
+use crate::cipher::AesCipher;
+use crate::db;
+use bitcoin::hashes::hex::ToHex;
+use bitcoin::secp256k1::SecretKey;
+use bitcoin::BlockHash;
+use bitcoin::Network;
+use lightning::chain::channelmonitor::ChannelMonitor;
+use lightning::sign::EntropySource;
+use lightning::sign::SignerProvider;
+use lightning::util::persist::KVStorePersister;
+use lightning::util::ser::Writeable;
+use lightning_persister::FilesystemPersister;
+use ln_dlc_node::storage::LDKStoreReader;
+use ln_dlc_node::storage::TenTenOneStorage;
+use ln_dlc_storage::sled::SledStorageProvider;
+use ln_dlc_storage::DLCStoreProvider;
+use std::fs;
+use std::ops::Deref;
+use std::path::Path;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct TenTenOneNodeStorage {
+    pub client: RemoteBackupClient,
+    pub ln_storage: Arc<FilesystemPersister>,
+    pub dlc_storage: Arc<SledStorageProvider>,
+    pub data_dir: String,
+    pub backup_dir: String,
+    pub network: Network,
+}
+
+impl TenTenOneStorage for TenTenOneNodeStorage {}
+
+impl TenTenOneNodeStorage {
+    pub fn new(data_dir: String, network: Network, secret_key: SecretKey) -> TenTenOneNodeStorage {
+        let mut data_dir = PathBuf::from(data_dir);
+        data_dir.push(network.to_string());
+
+        if !data_dir.exists() {
+            fs::create_dir_all(data_dir.as_path())
+                .unwrap_or_else(|_| panic!("Failed to create data dir {data_dir:?}"));
+        }
+
+        let backup_dir = data_dir.join(Path::new("backup"));
+        if !backup_dir.exists() {
+            fs::create_dir_all(backup_dir.as_path())
+                .unwrap_or_else(|_| panic!("Failed to create backup dir {backup_dir:?}"));
+        }
+
+        let data_dir = data_dir.to_string_lossy().to_string();
+        let backup_dir = backup_dir.to_string_lossy().to_string();
+        tracing::info!("Created backup dir at {backup_dir}");
+
+        let ln_storage = Arc::new(FilesystemPersister::new(data_dir.clone()));
+        let dlc_storage = Arc::new(SledStorageProvider::new(&data_dir));
+        let client = RemoteBackupClient::new(AesCipher::new(secret_key));
+
+        TenTenOneNodeStorage {
+            ln_storage,
+            dlc_storage,
+            data_dir,
+            backup_dir,
+            network,
+            client,
+        }
+    }
+
+    /// Creates a full backup of the lightning and dlc data.
+    pub async fn full_backup(&self) -> anyhow::Result<()> {
+        tracing::info!("Running full backup");
+        let mut handles = vec![];
+
+        let db_backup = db::backup()?;
+        let value = fs::read(db_backup)?;
+        let handle = self.client.backup("10101/db".to_string(), value);
```

Comment:
We should use constants for all those keys. I.e. `10101/db` here.

Reply:
bffe6b0591999a6dea36683e96d65014113e3e00